File: Examen_Unidad2/src/Visitante.java
public class Visitante {
    private static int counter = 1;
    private String id;
    private String nombre;
    private String apellidos;
    private String fechaNacimiento;
    private String curp;

    public Visitante(String nombre, String apellidos, String fechaNacimiento, String curp) {
        this.id = "V" + counter++;
        this.nombre = nombre;
        this.apellidos = apellidos;
        this.fechaNacimiento = fechaNacimiento;
        this.curp = curp;
    }

    public String getId() {
        return id;
    }

    public String getNombre() {
        return nombre;
    }

    public void setNombre(String nombre) {
        this.nombre = nombre;
    }

    public String getApellidos() {
        return apellidos;
    }

    public void setApellidos(String apellidos) {
        this.apellidos = apellidos;
    }

    public String getFechaNacimiento() {
        return fechaNacimiento;
    }

    public void setFechaNacimiento(String fechaNacimiento) {
        this.fechaNacimiento = fechaNacimiento;
    }

    public String getCurp() {
        return curp;
    }

    public void setCurp(String curp) {
        this.curp = curp;
    }

    @Override
    public String toString() {
        return "Visitante" +
                "Id:'" + id + '\'' +
                ", Nombre:'" + nombre + '\'' +
                ", Apellidos:'" + apellidos + '\'' +
                ", Fecha Nacimiento:'" + fechaNacimiento + '\'' +
                ", Curp:'" + curp + '\'';
    }
}

File: Examen_Unidad2 py/ZOO PY/Visitante.py
class Visitante:
  counter = 1

  def __init__(self, nombre, apellidos, fechaNacimiento, curp):
      self.id = "V" + str(Visitante.counter)
      Visitante.counter += 1
      self.nombre = nombre
      self.apellidos = apellidos
      self.fechaNacimiento = fechaNacimiento
      self.curp = curp

  def getId(self):
      return self.id

  def getNombre(self):
      return self.nombre

  def setNombre(self, nombre):
      self.nombre = nombre

  def getApellidos(self):
      return self.apellidos

  def setApellidos(self, apellidos):
      self.apellidos = apellidos

  def getFechaNacimiento(self):
      return self.fechaNacimiento

  def setFechaNacimiento(self, fechaNacimiento):
      self.fechaNacimiento = fechaNacimiento

  def getCurp(self):
      return self.curp

  def setCurp(self, curp):
      self.curp = curp

  def __str__(self):
      return f"Visitante: Id: '{self.id}', Nombre: '{self.nombre}', Apellidos: '{self.apellidos}', Fecha Nacimiento: '{self.fechaNacimiento}', Curp: '{self.curp}'"
